src/indexClinicalData/transformToEsDonor.ts

```diff
@@ -52,11 +53,43 @@ export default (
     submittedExtendedDataPercent: submittedExtendedDataPercent,
 
     registeredNormalSamples: donor.specimens
-      .filter((specimen) => specimen.tumourNormalDesignation === "Normal")
+      .filter(
+        (specimen) =>
+          specimen.tumourNormalDesignation === "Normal" &&
+          specimen.samples[0]?.sampleType
+            .toUpperCase()
+            .includes(DNA_SAMPLE_TYPE_KEYWORD)
+      )
       .reduce((sum, specimen) => sum + specimen.samples.length, 0),
 
     registeredTumourSamples: donor.specimens
-      .filter((specimen) => specimen.tumourNormalDesignation === "Tumour")
+      .filter(
+        (specimen) =>
+          specimen.tumourNormalDesignation === "Tumour" &&
+          specimen.samples[0]?.sampleType
+            .toUpperCase()
+            .includes(DNA_SAMPLE_TYPE_KEYWORD)
+      )
+      .reduce((sum, specimen) => sum + specimen.samples.length, 0),
+
+    rnaRegisteredNormalSamples: donor.specimens
+      .filter(
+        (specimen) =>
+          specimen.tumourNormalDesignation === "Normal" &&
+          specimen.samples[0]?.sampleType
+            .toUpperCase()
+            .includes(RNA_SAMPLE_TYPE_KEYWORD)
+      )
+      .reduce((sum, specimen) => sum + specimen.samples.length, 0),
+
+    rnaRegisteredTumourSamples: donor.specimens
+      .filter(
+        (specimen) =>
+          specimen.tumourNormalDesignation === "Tumour" &&
+          specimen.samples[0]?.sampleType
+            .toUpperCase()
+            .includes(RNA_SAMPLE_TYPE_KEYWORD)
+      )
       .reduce((sum, specimen) => sum + specimen.samples.length, 0),
```

Comment:
All of this is fine but is a lot of repeated code... Maybe you want a function that takes the `tumorNormalDesignation` value and the `sampleType` and counts for those:

```
function filterAndCountSamples(specimens: Specimen[], tumourNormalDesignation: "Tumour" | "Normal", sampleType: RNA_SAMPLE_TYPE_KEYWORD | DNA_SAMPLE_TYPE_KEYWORD): number {
  return specimens.filter((specimen) =>
          specimen.tumourNormalDesignation === tumourNormalDesignation &&
          specimen.samples[0]?.sampleType
            .toUpperCase()
            .includes(sampleType)
  ).reduce((sum, specimen) => sum + specimen.samples.length, 0);
}

Reply:
thanks!